### PR TITLE
Mechanism updated based on AA learning

### PR DESCRIPTION
**Pull request by Tan560**

### CODE REVIEW

File: src/main/cpp/mechanisms/ClimberManager/decoratormods/autoClimbState.cpp [hunk 1]
```diff
@@ -50,31 +51,37 @@ void autoClimbState::Init()
...
-			m_mechanism->UpdateTarget(RobotElementNames::MOTOR_CONTROLLER_USAGE::CLIMBER_MANAGER_RIGHT_CLIMBER, m_target);
-			m_mechanism->UpdateTarget(RobotElementNames::MOTOR_CONTROLLER_USAGE::CLIMBER_MANAGER_LEFT_CLIMBER, m_target);
-			m_bottomReached = true;
+			m_target = 8.5;
 		}
-	}*/
+	}
 	if (abs(TeleopControl::GetInstance()->GetAxisValue(TeleopControlFunctions::MANUAL_CLIMB)) > 0.05)
 	{
 		double delta = 6.0 * 0.035 * (TeleopControl::GetInstance()->GetAxisValue(TeleopControlFunctions::MANUAL_CLIMB)); // changing by 6 in/s * 0.05 for 20 ms loop time * controller input
```
Comment (witcpalek) [inline src/main/cpp/mechanisms/ClimberManager/decoratormods/autoClimbState.cpp:74]: comment and code seem out of sync  

Also there are a bunch of "magic" numbers that we might want to consider creating constants for

File: src/main/cpp/mechanisms/noteManager/decoratormods/noteManager.cpp [hunk 2]
```diff
@@ -233,19 +236,33 @@ void noteManager::Update(RobotStateChanges::StateChange change, int value)
...
 	{
 		distanceFromTarget = GetVisionDistance().to<double>();
+	}
+	else if (chassis != nullptr)
+	{
+		fieldElementPose = allianceColor == frc::DriverStation::Alliance::kRed ? frc::Pose3d{FieldConstants::GetInstance()->GetFieldElement(FieldConstants::FIELD_ELEMENT::RED_SPEAKER)} /*load red speaker*/ : frc::Pose3d{FieldConstants::GetInstance()->GetFieldElement(FieldConstants::FIELD_ELEMENT::BLUE_SPEAKER)}; /*load blue speaker*/
+		chassisPos = chassis->GetPose();
 
-		launchAngle = 80.0 + (-44.2 * distanceFromTarget) + (6.09 * distanceFromTarget * distanceFromTarget);
+		distanceFromTarget = sqrt(pow((fieldElementPose.X() - chassisPos.X()).to<double>(), 2) + pow((fieldElementPose.Y() - chassisPos.Y()).to<double>(), 2));
```
Comment (witcpalek) [inline src/main/cpp/mechanisms/noteManager/decoratormods/noteManager.cpp:256]: Why aren't you just using the WPILIB methods?

distanceFromTarget = (fieldElementPose.ToPose2d()-chassisPos).Translation().Distance();

File: src/main/cpp/mechanisms/noteManager/decoratormods/noteManager.cpp [hunk 3]
```diff
@@ -233,19 +236,33 @@ void noteManager::Update(RobotStateChanges::StateChange change, int value)
...
 	}
+
+	launchAngle = 80.0 + (-44.2 * distanceFromTarget) + (6.09 * distanceFromTarget * distanceFromTarget);
```
Comment (witcpalek) [inline src/main/cpp/mechanisms/noteManager/decoratormods/noteManager.cpp:259]: Can we replace the magic numbers with constants to make this easier to read?

**Review by witcpalek — CHANGES_REQUESTED** [review]

File: src/main/cpp/mechanisms/noteManager/decoratormods/noteManager.cpp (hunk 3 above)
Comment (Tan560) [reply src/main/cpp/mechanisms/noteManager/decoratormods/noteManager.cpp:259]: Leaving until we make a new function

File: src/main/cpp/mechanisms/noteManager/decoratormods/noteManager.cpp (hunk 2 above)
Comment (Tan560) [reply src/main/cpp/mechanisms/noteManager/decoratormods/noteManager.cpp:256]: Agreed to leave it the way it is

File: src/main/cpp/mechanisms/ClimberManager/decoratormods/autoClimbState.cpp (hunk 1 above)
Comment (Tan560) [reply src/main/cpp/mechanisms/ClimberManager/decoratormods/autoClimbState.cpp]: Updated comment to match

**Review by witcpalek — APPROVED** [review]

**Pull request merged**